heon/generate_flow_seq.py

```diff
@@ -0,0 +1,111 @@
+import argparse
+import random
+
+
+def flush_file(file):
+    f = open(file, "w")
+    f.write("")
+    f.close()
+
+
+def write_flow_num_to_file(file, value):
+    f = open(file, "a")
+    f.write(f"{value}\n")
+    f.close()
+
+
+def run_single_iteration(flow_num_list, file_to_write):
+    list_to_choose_from = flow_num_list.copy()
+    for i in range(len(list_to_choose_from)):
+        chosen_num = random.choice(list_to_choose_from)
+        write_flow_num_to_file(file_to_write, chosen_num)
+        list_to_choose_from.remove(chosen_num)
+
+
+def generate_random_flow_sequence_grouped(flow_num_list, num_iter, file_to_write):  
+    for i in range(0, num_iter):
+        run_single_iteration(flow_num_list, file_to_write)
+        
+
+def generate_random_flow_sequence_independent(num_conn, flow_num_list, num_iter, file_to_write):
+    flow_num_remaining_count_dict = {}
+    for i in range(1, num_conn+1):
+        flow_num_remaining_count_dict[i] = num_iter
+        
+    list_to_choose_from = flow_num_list.copy()
+    for i in range(0, num_conn * num_iter):
+        chosen_num = random.choice(list_to_choose_from)
+        write_flow_num_to_file(file_to_write, chosen_num)
+        remaining_count = flow_num_remaining_count_dict[chosen_num]
+        remaining_count = remaining_count - 1
+        flow_num_remaining_count_dict[chosen_num] = remaining_count
+        if remaining_count <= 0:
+            list_to_choose_from.remove(chosen_num)
```

Comment:
it's pretty cool that you maintain the invariant that there are still `num_iter` packets of each flow in your final list.

Reply:
Are you talking about `flow_num_remaining_count_dict`?